sql/core/src/test/scala/org/apache/spark/sql/execution/joins/BroadcastJoinSuite.scala

```diff
@@ -110,15 +110,39 @@ class BroadcastJoinSuite extends QueryTest with SQLTestUtils {
   }
 
   test("broadcast hint is retained after using the cached data") {
-    withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1") {
-      val df1 = spark.createDataFrame(Seq((1, "4"), (2, "2"))).toDF("key", "value")
-      val df2 = spark.createDataFrame(Seq((1, "1"), (2, "2"))).toDF("key", "value")
-      df2.cache()
-      val df3 = df1.join(broadcast(df2), Seq("key"), "inner")
-      val numBroadCastHashJoin = df3.queryExecution.executedPlan.collect {
-        case b: BroadcastHashJoinExec => b
-      }.size
-      assert(numBroadCastHashJoin === 1)
+    try {
+      withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1") {
+        val df1 = spark.createDataFrame(Seq((1, "4"), (2, "2"))).toDF("key", "value")
+        val df2 = spark.createDataFrame(Seq((1, "1"), (2, "2"))).toDF("key", "value")
+        df2.cache()
+        val df3 = df1.join(broadcast(df2), Seq("key"), "inner")
+        val numBroadCastHashJoin = df3.queryExecution.executedPlan.collect {
+          case b: BroadcastHashJoinExec => b
+        }.size
+        assert(numBroadCastHashJoin === 1)
+      }
+    } finally {
+      spark.catalog.clearCache()
```

Comment:
We have to clear the cache. We have another bug to fix in the cache. Will submit another PR to fix the cache plan matching.

Reply:
If we have to clear the cache, can we add `clearCache` into `afterEach` in general instead of adding this case-by-case?

Reply:
This test suite is not for testing cache. Thus, it is fine to do it for these two test cases.

Reply:
do you mean we can remove the cache cleaning here after you fix that bug?

Reply:
Yeah. That should be a separate bug.